polaris-react/src/components/Scrollable/Scrollable.tsx

```diff
@@ -36,218 +32,86 @@ export interface ScrollableProps extends React.HTMLProps<HTMLDivElement> {
   onScrolledToBottom?(): void;
 }
 
-interface State {
-  topShadow: boolean;
-  bottomShadow: boolean;
-  scrollPosition: number;
-  canScroll: boolean;
-}
-
-export class Scrollable extends Component<ScrollableProps, State> {
-  static ScrollTo = ScrollTo;
-  static forNode(node: HTMLElement): HTMLElement | Document {
-    const closestElement = node.closest(scrollable.selector);
-    return closestElement instanceof HTMLElement ? closestElement : document;
-  }
-
-  state: State = {
-    topShadow: false,
-    bottomShadow: false,
-    scrollPosition: 0,
-    canScroll: false,
-  };
-
-  private stickyManager = new StickyManager();
-
-  private scrollArea: HTMLElement | null = null;
-
-  private handleResize = debounce(
-    () => {
-      this.handleScroll();
-    },
-    50,
-    {trailing: true},
-  );
-
-  componentDidMount() {
-    if (this.scrollArea == null) {
-      return;
-    }
-    this.stickyManager.setContainer(this.scrollArea);
-    this.scrollArea.addEventListener('scroll', () => {
-      window.requestAnimationFrame(this.handleScroll);
-    });
-    window.addEventListener('resize', this.handleResize);
-    window.requestAnimationFrame(() => {
-      this.handleScroll();
-      if (this.props.hint) {
-        this.scrollHint();
-      }
-    });
-  }
-
-  componentWillUnmount() {
-    if (this.scrollArea == null) {
-      return;
+export function Scrollable({
+  children,
+  className,
+  horizontal,
+  vertical = true,
+  shadow,
+  hint,
+  focusable,
+  onScrolledToBottom,
+  ...rest
+}: ScrollableProps) {
+  const [topShadow, setTopShadow] = useState(false);
+  const [bottomShadow, setBottomShadow] = useState(false);
+  const stickyManager = useRef(new StickyManager());
+  const scrollArea = useRef<HTMLDivElement>(null);
+  const scrollTo = useCallback((scrollY: number) => {
+    scrollArea.current?.scrollTo({top: scrollY, behavior: 'smooth'});
+  }, []);
+
+  useEffect(() => {
+    if (hint) {
```

Comment:
I considered this as well. Opted to change because:
- from a maintenance perspective, trying to avoid the cost of adding something like `useEffectOnce` (to mimic componentDidMount) or disabling exhaustive deps
- it could be desired behavior for a consumer to toggle the prop and trigger the hint sometime after initial render

That said, it's not 1:1 with the old. It's probably more responsible to make it 1:1 (as much as possible) unless we want to push the new version higher than patch.

Reply:
> That said, it's not 1:1 with the old. It's probably more responsible to make it 1:1 (as much as possible) unless we want to push the new version higher than patch.

That's fair. I'm fine with either. It does seem a little odd to only use the value on mount, since that's not reactive. But it was the previous implementation 🤷